app/controllers/admin/discourse_calendar/admin_holidays_controller.rb

```diff
@@ -1,19 +1,39 @@
 # frozen_string_literal: true
 
-require "holidays"
-
 module Admin::DiscourseCalendar
   class AdminHolidaysController < AdminDiscourseCalendarController
     def index
       region_code = params[:region_code]
 
       begin
-        holidays = Holidays.year_holidays([region_code], Time.current.beginning_of_year)
+        holidays = DiscourseCalendar::Holiday.find_holidays_for(region_code: region_code)
       rescue Holidays::InvalidRegion
         return render_json_error(I18n.t("system_messages.discourse_calendar_holiday_region_invalid"), 422)
       end
 
       render json: { region_code: region_code, holidays: holidays }
     end
+
+    def disable
+      DiscourseCalendar::DisabledHoliday.create(disabled_holiday_params)
+      CalendarEvent.delete_by(
```

Comment:
```suggestion
      CalendarEvent.destroy_by(
```

`#delete_by` doesn't run callbacks. Should we check what happens if there's no `CalendarEvent`?

Reply:
I updated this to use `destroy_by`. I was thinking about checking if a calendar event was deleted or not, but I can't seem to find a reason to check for that. One thought came to me though, maybe if the UI alerted a message if an event was deleted or not. Wondering if you had something in mind for this? :thinking: :smile: